### PR TITLE
Add Celestica device platform of Questone2/2A

### PR DESCRIPTION
**Pull request by Rancho333**
Signed-off-by: Rancho zhwwan@gmail.com

#### Why I did it
Add new device  Questone2/2A of network switch platform with Broadcom ASIC.

#### How I did it
- Add device-specific plugins and config.
- Add platform modules

#### How to verify it
- Build a Broadcom SONiC image.
- Show version

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- 202111
- [ ] 202205

#### Description for the changelog
Add Celestica device platform of Questone2/2A.

### CODE REVIEW

**Comment by Rancho333** [comment]
> This pull request **introduces 1 alert** when merging [c5456eb](https://github.com/Azure/sonic-buildimage/commit/c5456eb6035a4685c3955d1651a65875a41b6dd3) into [e8daeac](https://github.com/Azure/sonic-buildimage/commit/e8daeacd3ddcfd4fa3e01fa890aa65cd51934712) - [view on LGTM.com](https://lgtm.com/projects/g/Azure/sonic-buildimage/rev/pr-9e4d4e55ba0f47dd364b1fc0d03361a03410502f)
> 
> **new alerts:**
> 
> * 1 for Wrong number of arguments in a class instantiation

**Comment by Rancho333** [comment]
@lguohan Can you review this PR? Thanks!

File: device/celestica/x86_64-cel_questone_2-r0/Questone_2/port_config.ini [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# name      lanes            alias   index  speed   mtu   admin_status  fec
+Ethernet1   49               25GE1   0      25000   9216  up            rs
```
Comment (Blueve) [inline device/celestica/x86_64-cel_questone_2-r0/Questone_2/port_config.ini:2]: According to this doc, alias naming should follow `etpN` where N>=1

**Comment by Blueve** [comment]
@sujinmkang could you help review this PR?

File: device/celestica/x86_64-cel_questone_2-r0/Questone_2/port_config.ini (hunk 1 above)
Comment (Rancho333) [reply device/celestica/x86_64-cel_questone_2-r0/Questone_2/port_config.ini:2]: Hi, Blueve, thanks for you review. Is this necessary? It's just an alias. I saw different manufacturers have different  alias styles.

**Comment by zhangyanzhao** [comment]
@Rancho333  please create another PR to master branch to merge, then we will cherry pick to 202111. Usually, we do not directly merge to sub branch. I will close this PR.

**Pull request closed without merging**